src/stub_launcher.cc

```diff
@@ -442,7 +442,11 @@ StubLauncher::Launch()
         stub_message_queue_.reset();
         parent_message_queue_.reset();
         memory_manager_.reset();
-        WaitForStubProcess();
+        if (stub_pid_ != 0) {
```

Comment:
Question for my own curiosity @indrajit96 @Tabrizian @krishung5: 

Are the other spots that call `WaitForStubProcess()` not at risk for similar issues?

Does it make sense to just move this if statement within `WaitForStubProcess()` instead? Should it ever wait on PID 0?

Reply:
Great question! I think usually `WaitForStubProcess()` comes with killing the stub process, so it's the error handling cases that we need to be cautious. I think it makes sense to move the check within `WaitForStubProcess()`. The stub pid is reset to 0 when it's killed, so it shouldn't be waiting on PID 0.

Reply:
+1, Will move it inside and test once